tests/generic_config_updater/test_syslog.py

```diff
@@ -0,0 +1,265 @@
+import logging
+import pytest
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.config_reload import config_reload
+from tests.generic_config_updater.gu_utils import apply_patch, expect_op_success_and_reset_check, expect_res_success, expect_op_failure
+from tests.generic_config_updater.gu_utils import generate_tmpfile, delete_tmpfile
+
+pytestmark = [
+    pytest.mark.topology('any'),
+]
+
+logger = logging.getLogger(__name__)
+
+SYSLOG_THRESHOLD=10
+SYSLOG_INTERVAL=1
+# This is restricted by sonic-syslog.yang. Use '-1' to indicate no max is set
+SYSLOG_MAX_SERVER=-1
+# The max server test only support SYSLOG_MAX_SERVER that is equal or lower than 254.
+# Will modify if change in future
+SYSLOG_TEST_MAX_UPPER_LIMIT=254
+
+@pytest.fixture(scope="module")
+def setup_env(duthosts, rand_one_dut_hostname, cfg_facts):
+    """
+    Setup/teardown fixture for syslog config
+    Args:
+        duthosts: list of DUTs.
+        rand_selected_dut: The fixture returns a randomly selected DuT.
+        cfg_facts: config facts for selected DUT
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+
+    config_tmpfile = generate_tmpfile(duthost)
+    logger.info("config_tmpfile {} Backing up config_db.json".format(config_tmpfile))
+    duthost.shell("sudo cp /etc/sonic/config_db.json {}".format(config_tmpfile))
+
+    # Cleanup syslog server config
+    syslog_servers = cfg_facts.get('SYSLOG_SERVER', {})
+    for syslog_server in syslog_servers:
+        del_syslog_server = duthost.shell("sudo config syslog del {}".format(syslog_server),
+            module_ignore_errors=True)
+        pytest_assert(not del_syslog_server['rc'],
+            "syslog server '{}' is not deleted successfully".format(syslog_server))
+
+    yield
+
+    logger.info("Restoring config_db.json")
+    duthost.shell("sudo cp {} /etc/sonic/config_db.json".format(config_tmpfile))
```

Comment:
>cp [](http://example.com/codeflow?start=24&length=2)

We also need to restore the Redis database in memory with command line `sudo config reload -y`.